_security/access-control/api.md

```diff
@@ -1297,6 +1297,91 @@ PATCH _plugins/_security/api/securityconfig
 }
 ```
 
+### Configuration upgrade check
+
+Introduced 2.13
+{: .label .label-purple }
+
+Checks the cuurent configuration bundled with the host's Security plugin and compares it to the latest Security plugin bundled with OpenSearch. Then, the API returns whether or not can be performed and what resources can be updated.
```

Comment:
There appear to be some words missing in the second sentence.

Reply:
> Original text: Checks the configuration bundled with the Security plugin against the existing security configuration, returns if an upgrade can be performed and what resources would be updated.

Here is a diagram of what the actual flow / comparisons are. (Thanks ChatGPT!)

```mermaid
sequenceDiagram
    participant U as User
    participant D as Configuration on Disk
    participant S as SecurityPlugin
    participant M as Configuration in SecurityPlugin Memory
    U->>+S: API Call to Check Configuration
    S->>+D: Retrieve Current Configuration from Disk
    D-->>-S: Current Disk Configuration
    S->>+M: Check Configuration in Memory
    M-->>-S: Current in Memory Configuration
    S->>S: Compare Configurations
    alt Configurations Match
        S-->>U: No Update Needed
    else Configurations Do Not Match
        S->>S: Determine Resources for Update
        S-->>U: Update Can be Performed, List resources to Update
    end

```

Reply:
I feel like that original text might be a better starting point, but let me know if you'd like me to workshop it more

Reply:
@peternied: I think I'm confused as to what you mean by "bundled"? When I hear bundled, I assume the API is checking for upgrades either against the upstream instance (The newest artifact) or the version the artifact the user downloaded. But, if the user has to download a new artifact for the latest Security Plugin information to persist in memory, isn't there a step we're missing in the documentation then?

Reply:
I mean `the version the artifact the user downloaded`.

This API is for after an upgrade was performed, such as a [Rolling Upgrade](https://opensearch.org/docs/latest/install-and-configure/upgrade-opensearch/rolling-upgrade/).

Note; I think this highlights another part of the documentation that I should include 'now that you've upgraded you can [optionally] upgrade the security configuration via this API' which links to this documentation